Add login-forms module

chillu opened the pull request:
We created this for the MFA module, so it's a supported module.
Doesn't have a stable release yet, hence the @beta tag.

Depends on https://github.com/silverstripeltd/open-sourcerers/issues/68
Related to https://github.com/silverstripe/silverstripe-login-forms/pull/32

chillu committed:
Add login-forms module

We created this for the MFA module, so it's a supported module.
Doesn't have a stable release yet, hence the @beta tag.

chillu committed:
Login form compat with modal dialog

See https://github.com/silverstripeltd/open-sourcerers/issues/68

maxime-rainville commented on app/templates/Security.ss:
```diff
@@ -0,0 +1,64 @@
+        <footer class="silverstripe-brand">
+            <% include SilverStripeLogo %>
+        </footer>
+        $ModalWindow
```
Originally, I thought about hardcoding the messages into the the JS file and adding the modal to the DOM with JQuery. I opted against it thinking it was too hackish.

But if it saves us from overriding an entire template, it might be worth reconsidering.

maxime-rainville reviewed (changes requested):
I don't seem to be hitting the  `app/templates/Security.ss` template locally.

You still need to require `app/css/dialog.css` and `app/js/dialog.js`. Otherwise there's nothing to trigger the modal display.

chillu committed:
Updated login-forms to 4.0.0-beta2

chillu committed:
Set root path on cookie, avoid dialog on different paths

chillu commented:
> I don't seem to be hitting the app/templates/Security.ss template locally.

That's because my PR to login-forms was merged, but it wasn't pulling in the changes because it's pinned to `@beta`. I've created [4.0.0-beta2](https://github.com/silverstripe/silverstripe-login-forms/releases/tag/4.0.0-beta2) now and updated composer.lock.

> You still need to require app/css/dialog.css and app/js/dialog.js. Otherwise there's nothing to trigger the modal display.

`Security/*` routes are still rendered through a fake `PageController`, which means `init()` gets called, and those requirements are included. I think this is also resolved by the beta2 release above. Since it wasn't using the `app/` template, it didn't include jQuery, so the modal dialog JS would've caused JS errors.

When changing the cookie logic to JS only, you didn't add the path - so the dialog showed on different paths and set path specific cookies. I've fixed that as part of this PR as well.

maxime-rainville reviewed (changes requested):
I'm seeing the modal now ... however the style is substantially different from the regular page.

![image](https://user-images.githubusercontent.com/1168676/60852504-111f7800-a24c-11e9-8516-dccfd5ca8e63.png)
![image](https://user-images.githubusercontent.com/1168676/60852511-17adef80-a24c-11e9-89d0-a0c27a8181d3.png)

I think behat tests are failing because Behat doesn't know how to click the "I understand" button. :laughing:

chillu commented:
> I think behat tests are failing because Behat doesn't know how to click the "I understand" button.

OK, so initially it was "cheap" to add Behat builds (just running `admin` tests), now it's expensive (customisations). I'm recommending that we disable them for now: https://github.com/silverstripe/bambusa-installer/issues/21

chillu commented:
The problem is that login-forms redefines the root `font-size` to `13px`. @clarkepaul How important is it to fix that? They're both legible, and you only see one or the other, never both variations.

![image](https://user-images.githubusercontent.com/111025/60925336-7b412700-a2f7-11e9-9a5f-fe705f33ee89.png)

maxime-rainville commented:
I'm not sure how behat will handle the new login-form. If it's broken, we'll need to fix it upstream any way because we want to add the login-form to our base installer.

Once https://github.com/silverstripe/bambusa-installer/pull/23 gets merged, I'll rebase this PR and figure out if behat plays nice with the new login form.

Commit:
Updated composer.lock

Commit:
Uniformise the look of the modal between regular site and login page

maxime-rainville reviewed (approved):
Behat is happy with the new login page.

I've squash the difference in the modal between the login page and the refular site 
![image](https://user-images.githubusercontent.com/1168676/60931583-2361ea80-a30e-11e9-9a31-d197a1a0268f.png) 
![image](https://user-images.githubusercontent.com/1168676/60931593-2e1c7f80-a30e-11e9-855e-4659ef75bda7.png)

Updated the message and remove the button border as discussed here https://github.com/silverstripeltd/open-sourcerers/issues/68#issuecomment-509474599

maxime-rainville commented:
Also had a quick look in Chrome, Firefox and IE11 to make it sure it looks suitable in all browsers.

clarkepaul commented:
I didn't see much of a noticeable change of the font size so not something that I'm pushing.

maxime-rainville commented:
I fixed the font size already. I had to fix the to switch from `rem` to `em` for this to work.
